src/v/raft/consensus.cc

```diff
@@ -1500,7 +1500,7 @@ ss::future<> consensus::write_last_applied(model::offset o) {
      * In order to keep an invariant that: 'last applied offset MUST be
      * readable' we limit it here to committed (leader flushed) offset.
      */
-    auto const limited_offset = std::min(o, _log->offsets().committed_offset);
+    auto const limited_offset = std::min(o, _flushed_offset);
```

Comment:
Just curious, why this instead of `_commit_index`?

Also, is there a description of `_flushed_offset` anywhere?

Reply:
My understanding is that this is committed offset in the storage sense - i.e. locally flushed (OTOH committed in the raft sense == replicated and flushed on the majority of nodes). The idea of `_flushed_offset` is that raft is responsible for tracking the last offset it flushed itself (instead of relying on the log to track it internally). This way (if log clients track the flushed offset externally) we can get rid of internal tracking (`_log->offsets().committed_offset`)

Reply:
exactly

Reply:
Thanks! Makes sense. I wasn't sure if this change was using `_flushed_offset` because that was always the correct thing to do, or if this was explicitly a step away from using `offsets().committed_offset`).